test/io/write_sqd_sim_result.cpp

```diff
@@ -0,0 +1,175 @@
+{
+    using namespace std::chrono_literals;
+
+    using sidb_layout = cell_level_layout<sidb_technology, clocked_layout<cartesian_layout<siqad::coord_t>>>;
+
+    sidb_layout lyt{{20, 10}};
+
+    lyt.assign_cell_type({0, 0, 0}, sidb_layout::cell_type::NORMAL);
+    lyt.assign_cell_type({5, 0, 0}, sidb_layout::cell_type::NORMAL);
+    lyt.assign_cell_type({7, 0, 0}, sidb_layout::cell_type::NORMAL);
+
+    lyt.assign_cell_type({11, 0, 0}, sidb_layout::cell_type::NORMAL);
+    lyt.assign_cell_type({13, 0, 0}, sidb_layout::cell_type::NORMAL);
+
+    lyt.assign_cell_type({17, 0, 0}, sidb_layout::cell_type::NORMAL);
+    lyt.assign_cell_type({19, 0, 0}, sidb_layout::cell_type::NORMAL);
+
+    exgs_stats<sidb_layout>          exgs_stats{};
+    const sidb_simulation_parameters params{2, -0.32};
+
+    exhaustive_ground_state_simulation<sidb_layout>(lyt, params, &exgs_stats);
+
+    sidb_simulation_result<sidb_layout> sim_result{};
+
+    sim_result.algorithm_name       = "ExGS";
+    sim_result.simulation_runtime   = exgs_stats.time_total;
+    sim_result.charge_distributions = exgs_stats.valid_lyts;
+    sim_result.physical_parameters  = params;
+
+    std::stringstream simulation_stream{};
+
+    const std::string sim_result_str = fmt::format(
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<sim_out>\n"
+        "    <eng_info>\n"
+        "        <engine>ExGS</engine>\n"
+        "        <version>{}</version>\n"
+        "        <repo>{}</repo>\n"
+        "        <return_code>0</return_code>\n"
+        "        <timestamp>{}</timestamp>\n"
+        "        <time_elapsed_s>{}</time_elapsed_s>\n"
+        "    </eng_info>\n"
+        "    <sim_params>\n"
+        "        <debye_length>{}</debye_length>\n"
+        "        <eps_r>{}</eps_r>\n"
+        "        <muzm>{}</muzm>\n"
+        "    </sim_params>\n"
+        "    <physloc>\n"
+        "        <dbdot x=\"0.0\" y=\"0.0\"/>\n"
+        "        <dbdot x=\"4.224e-09\" y=\"0.0\"/>\n"
+        "        <dbdot x=\"6.528000000000001e-09\" y=\"0.0\"/>\n"
+        "        <dbdot x=\"1.92e-09\" y=\"0.0\"/>\n"
+        "        <dbdot x=\"7.2960000000000004e-09\" y=\"0.0\"/>\n"
+        "        <dbdot x=\"4.992e-09\" y=\"0.0\"/>\n"
+        "        <dbdot x=\"2.6880000000000002e-09\" y=\"0.0\"/>\n"
+        "    </physloc>\n"
+        "    <elec_dist>\n"
+        "        <dist energy=\"0.24602741408534134\" count=\"1\" physically_valid=\"1\" "
+        "state_count=\"2\">0---000</dist>\n"
+        "    </elec_dist>\n"
+        "</sim_out>\n",
+        FICTION_VERSION, FICTION_REPO, fmt::format("{:%Y-%m-%d %H:%M:%S}", fmt::localtime(std::time(nullptr))),
+        sim_result.simulation_runtime.count(), sim_result.physical_parameters.lambda_tf,
+        sim_result.physical_parameters.epsilon_r, sim_result.physical_parameters.mu);
+
+    write_sqd_sim_result(sim_result, simulation_stream);
+
+    CHECK(simulation_stream.str() == sim_result_str);
+}
```

Comment:
Do you think we should add a test to cover positively charged DBs? Or is this redundant?
We could just put three DBs right next to each other.

Reply:
Sure, we can do that!